rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java

```diff
@@ -64,49 +63,125 @@ public Stream<SessionRepresentation> realmSessions(@QueryParam("type") @DefaultV
+            Stream<ClientIdSessionType> sessionIdStream = Stream.<ClientIdSessionType>builder().build();
+            if (type == ALL || type == REGULAR) {
+                final Map<String, Long> clientSessionStats = session.sessions().getActiveClientSessionStats(realm, false);
+                sessionIdStream = Stream.concat(sessionIdStream, clientSessionStats
+                        .keySet().stream().map(i -> new ClientIdSessionType(i, REGULAR)));
+            }
+            if (type == ALL || type == OFFLINE) {
+                sessionIdStream = Stream.concat(sessionIdStream, session.sessions().getActiveClientSessionStats(realm, true)
+                        .keySet().stream().map(i -> new ClientIdSessionType(i, OFFLINE)));
+            }
 
-        Stream<SessionRepresentation> result = sessionIdStream.flatMap((sessionId) -> {
-            ClientModel clientModel = realm.getClientById(sessionId.getClientId());
-            switch (sessionId.getType()) {
-                case REGULAR:
-                    return session.sessions().getUserSessionsStream(realm, clientModel)
-                            .map(s -> toUserSessionRepresentation(s, sessionId.getClientId(), REGULAR));
-                case OFFLINE:
-                    return session.sessions()
-                            .getOfflineUserSessionsStream(realm, clientModel, null, null)
-                            .map(s -> toUserSessionRepresentation(s, sessionId.getClientId(), OFFLINE));
+            Stream<SessionRepresentation> result = sessionIdStream.flatMap((clientIdSessionType) -> {
+                // TODO: Will not work if it is an external client IDs, see JpaUserSessionPersister#getUserSessionsCountsByClients
+                ClientModel clientModel = realm.getClientById(clientIdSessionType.getClientId());
+                switch (clientIdSessionType.getType()) {
+                    case REGULAR:
+                        return session.sessions().getUserSessionsStream(realm, clientModel)
+                                .map(s -> toRepresentation(s, REGULAR));
+                    case OFFLINE:
+                        return session.sessions()
+                                .getOfflineUserSessionsStream(realm, clientModel, null, null)
+                                .map(s -> toRepresentation(s, OFFLINE));
+                }
+                return Stream.<SessionRepresentation>builder().build();
+            }).distinct();
+
+            if (!search.equals("")) {
+                result = result.filter(s -> s.getUsername().contains(search) || s.getIpAddress().contains(search)
+                                            || s.getClients().values().stream().anyMatch(c -> c.contains(search)));
+            }
+            return result.skip(first).limit(max);
+        } else {
+            ClientModel client = null;
+            UserModel user = null;
+            String ipAddress;
+            if (!search.isEmpty()) {
```

Comment:
Should this be rather `StringUtil.isNotBlank` and using `search.trim()` for substring searches? (Similarly to existing code above)

Reply:
done